server/serializer/serializers.go

```diff
@@ -0,0 +1,105 @@
+package serializer
+
+import (
+	"net/http"
+
+	"github.com/src-d/code-annotation/server/model"
+)
+
+//TODO: Responses and Redirects should implement Serializable{Serialize(w)}
+
+type Response struct {
+	Status      int         `json:"status"`
+	Data        interface{} `json:"data,omitempty"`
+	Errors      []*Error    `json:"errors,omitempty"`
+	RedirectURL string
+}
+
+type Error struct {
+	Status  int    `json:"status"`
+	Title   string `json:"title"`
+	Details string `json:"details,omitempty"`
+}
+
+func newResponse(c interface{}) *Response {
+	if c == nil {
+		return &Response{
+			Status: http.StatusNoContent,
+		}
+	}
+
+	return &Response{
+		Status: http.StatusOK,
+		Data:   c,
+	}
+}
+
+func NewTemporalRedirect(URL string) *Response {
+	return &Response{Status: http.StatusTemporaryRedirect, RedirectURL: URL}
+}
+
+type ExperimentResponse struct {
+	ID          int    `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+func NewExperimentResponse(e *model.Experiment) *Response {
```

Comment:
you just broke name convention.
If a package has public `ExperimentResponse` structure, `NewExperimentResponse` should return it. Not `Response`.
Returning`Response` will make sense if `ExperimentResponse` is private.